src/service.rs

```diff
@@ -896,21 +1188,46 @@ impl Service {
     // Send RPC Requests //
 
     /// Sends a PING request to a node.
-    fn send_ping(&mut self, enr: Enr) {
-        match NodeContact::try_from_enr(enr, self.config.ip_mode) {
-            Ok(contact) => {
-                let request_body = RequestBody::Ping {
-                    enr_seq: self.local_enr.read().seq(),
-                };
-                let active_request = ActiveRequest {
-                    contact,
-                    request_body,
-                    query_id: None,
-                    callback: None,
-                };
-                self.send_rpc_request(active_request);
+    fn send_ping(&mut self, enr: &Enr, symmetric_port: Option<u16>, is_hole_punch: bool) {
+        if let Some(contact) = self.contact_from_enr(enr, symmetric_port) {
+            let request_body = RequestBody::Ping {
+                enr_seq: self.local_enr.read().seq(),
+            };
+            let active_request = ActiveRequest {
+                contact,
+                request_body,
+                query_id: None,
+                callback: None,
+                relay: None,
+            };
+            if is_hole_punch {
+                self.hole_punch_pings.insert(enr.node_id());
+                self.send_hole_punch_ping(active_request);
+            } else {
+                _ = self.send_rpc_request(active_request);
             }
-            Err(NonContactable { enr }) => error!("Trying to ping a non-contactable peer {}", enr),
+        }
+    }
+
+    /// Sends a PING request to a node.
+    fn send_hole_punch_ping(&mut self, active_request: ActiveRequest) {
```

Comment:
if this is just a ping, same as as with `send_ping` it should only receive the params needed for a ping. It's missleading being able to pass an `ActiveRequest` of any body type when this is not even going to be stored as an active request in `self.active_requests`